src/main.ts

```diff
@@ -1489,7 +1491,9 @@ function generateFromPartial(ctx: Context, fullName: string, messageDesc: Descri
             if (isBytes(valueType)) {
               return code`${from}`;
             } else if (isEnum(valueType)) {
-              return code`${from} as number`;
+              const mapType = detectMapType(ctx, messageDesc, field);
```

Comment:
Ah nice, @boukeversteegh this looks like a great fix; we were just hard-coded `as number` here before...makes sense.

Re-using `detectMapType` here made a lot of sense, but it made me curious why the code didn't already do that somewhere (given this all "handle maps" related code); I found this line:

```
const valueType = (typeMap.get(field.typeName)![2] as DescriptorProto).field[1];```

Which a) was doing basically what `detectMapType` itself does (but getting the `FieldDescriptorProto` instead of `detectMapType`'s output type), and b) was calling it `valueType` instead of what it probably should be called, `valueField`.

So I pushed up a change to this PR that does that, i.e. keeps your fix but removes the old `const valueType = ...ugly code...` with a call to `detectMapType` which now returns both `valueType` (as before) as well as `valueField`, which is what that line really wanted.

Reply:
Thanks a lot for that! I did notice that weird line, but didn't have the time to figure out if it was the same or not, so this is better for sure. Cheers!